oxidation/libparsec/crates/types/src/time.rs

```diff
@@ -37,22 +37,23 @@ pub use chrono::Duration; // Reexported
 pub struct DateTime(chrono::DateTime<chrono::Utc>);
 
 impl DateTime {
-    pub fn from_ymd_and_hms(
-        year: u64,
-        month: u64,
-        day: u64,
-        hour: u64,
-        minute: u64,
-        second: u64,
+    pub fn from_ymd_hms_us(
+        year: i32,
+        month: u32,
+        day: u32,
+        hour: u32,
+        minute: u32,
+        second: u32,
+        microsecond: u32,
     ) -> Self {
         Self(
             chrono::Utc
-                .ymd(year as i32, month as u32, day as u32)
-                .and_hms(hour as u32, minute as u32, second as u32),
+                .ymd(year, month, day)
+                .and_hms_micro(hour, minute, second, microsecond),
         )
     }
 
-    // Don't implement this as `From<f64>` to keep it private
+    // Don't implement this as `From<f64>` to prevent misunderstanding on precision
```

Comment:
```suggestion
    /// Don't implement this as `From<f64>` to prevent misunderstanding on precision.
```
This information would be useful as a doc-comment as it would be displayed during `hover` or `completion`